golang/trace/zipin_tracereporter_test.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package trace
+
+import (
+	"net"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/uber/tchannel/golang"
+	"github.com/uber/tchannel/golang/thrift"
+	gen "github.com/uber/tchannel/golang/trace/thrift/gen-go/tcollector"
+	"github.com/uber/tchannel/golang/trace/thrift/mocks"
+)
+
+func TestZipkinTraceReporterFactory(t *testing.T) {
+	_, err := tchannel.NewChannel("client", &tchannel.ChannelOptions{
+		Logger:               tchannel.SimpleLogger,
+		TraceReporterFactory: tchannel.TraceReporterFactory(ZipkinTraceReporterFactory),
+	})
+
+	assert.NoError(t, err)
+}
+
+func TestBuildZipkinSpan(t *testing.T) {
+	host := &tchannel.Endpoint{
+		Ipv4:        "127.0.0.1",
+		Port:        8888,
+		ServiceName: "test",
+	}
+	span := *tchannel.NewRootSpan()
+	name := "test"
+	annotations := RandomAnnotations()
+	thriftSpan := buildZipkinSpan(span, annotations, nil, name, host)
+
+	expectedSpan := gen.Span{
+		TraceId: uInt64ToBytes(span.TraceID()),
+		Host: &gen.Endpoint{
+			Ipv4:        (int32)(inetAton("127.0.0.1")),
+			Port:        8888,
+			ServiceName: "test",
+		},
+		Name:        name,
+		Id:          uInt64ToBytes(span.SpanID()),
+		ParentId:    uInt64ToBytes(span.ParentID()),
+		Annotations: buildZipkinAnnotations(annotations),
+		Debug:       false,
+	}
+	reflect.DeepEqual(thriftSpan, expectedSpan)
+}
+
+func TestInetAton(t *testing.T) {
+	assert.Equal(t, inetAton("1.0.0.1"), (uint32)(16777217))
```

Comment:
You don't need the parens around `uint32`. Also you should use a value like "1.2.3.4" since you want to make sure that the endianness being used is correct.
```go
assert.Equal(t, inetAton("1.2.3.4"), uint32(16909060))
```